C/common/reading.cpp

```diff
@@ -74,14 +74,42 @@ void Reading::addDatapoint(Datapoint *value)
 string Reading::toJSON()
 {
 ostringstream convert;
-
+char date_time[DATE_TIME_BUFFER_LEN];
+char micro_s[10];
 	convert << "{ \"asset_code\" : \"";
 	convert << m_asset;
 	convert << "\", \"read_key\" : \"";
 	convert << m_uuid;
 	convert << "\", \"user_ts\" : \"";
-	string ts = std::asctime(std::localtime(&m_timestamp));
-	convert << ts.substr(0, ts.length() - 1);
+
+	// Populate tm structure
+	const struct tm *timeinfo = std::localtime(&(m_timestamp.tv_sec));
+
+	/**
+	 * Build date_time with format YYYY-MM-DD HH24:MM:SS.MS+00:00
+	 * this is same as Python3:
+	 * datetime.datetime.now(tz=datetime.timezone.utc)
+	 */
+
+	// Create datetime with seconds
+	std::strftime(date_time, sizeof(date_time),
+		      DEFAULT_DATE_TIME_FORMAT,
+		      timeinfo);
+
+	// Add microseconds 
+        snprintf(micro_s,
```

Comment:
Don't really need the string manipulation stuff, this can be done in the C++ string and avoid some duplication

Reply:
usage of strftime and microseconds formatting will be part on next commits